### DISCUSSION TITLE
can modusa-ui be used as independent micro-front end framework as replacement of thymeleaf?

### DISCUSSION THREAD

**Discussion by muhammadismailkhan0009** [medusa-ui/medusa#402] [Q&A] [answered]
I plan to build my recent microservice projects as micro-front ends. 
I have hands-on experience in thymeleaf with spring boot mvc and spring reactive.
My question is: can i use modusa-ui as front-end framework as replacement of thymeleaf, and without using it with hydra?
The reason i don't want to hydra for now is that i want to learn all the features that hydra is providing by myself. 
So i am just looking for some micro-frontend framework.
Thank you

**Comment by kevindeyne** [comment, marked as answer]
Yes, as we're building this, we're trying to make sure developers can start out with just using single medusa-ui and can scale up to using hydra as the need arises (ie, they'll have multiple front-ends that want to work together). Just don't provide any of these properties:
```
medusa.hydra.uri
medusa.hydra.secret.public-key
medusa.hydra.secret.private-key
```
and it should start up in a local mode.

In fact, that is how our medusa showcase currently runs: https://medusa-showcase.onrender.com/

That does mean you'll have some more limited functionality, as hydra provisions the connection to other micro-frontends in the system. So without hydra, loading remote fragments will not be possible; as well as upcoming features like dynamic menus and such.
